### DISCUSSION TITLE
API usage?

### DISCUSSION THREAD

**Discussion by thomasazar** [formfcw/directus-extension-flexible-editor#58] [Q&A] [answered]
First off, wow! Thank you for the incredible work on this extension. I just happened across it in the Marketplace last week, and it looks fantastic. Once I wrapped my head around how to set it up correctly, adding new content became fun and easy.

What's throwing me for a loop is how to access that content. Say I've added a relation block for an image. It includes a caption, and the image file itself. How do I access that caption and filename through the Directus API? When I query the item with Flexible Editor, it only returns the UUID, junction collection name, and related collection name. I'm lost on how to actually retrieve the information, so that I could (for example) drop it into a statically-generated site.

**Comment by formfcw** [comment, marked as answer]
Thank you @thomasazar !
Have a look at Flexible Editor Content: https://github.com/formfcw/directus-extension-flexible-editor/tree/main/content
Basically, you would fetch the m2a field data and then inject it into the editor json.

**Comment by sekiranat** [comment]
Yes, indeed related fields can be retrieved like any other fields in directus. They can be highly nested. Don't forget to check read permissions.
